compiler/x/codegen/X86BinaryEncoding.cpp

```diff
@@ -948,6 +948,21 @@ TR::X86ImmInstruction::addMetaDataForCodeAddress(uint8_t *cursor)
                   (uint8_t *)(intptr_t)getNode()->getInlinedSiteIndex() , (TR_ExternalRelocationTargetKind) getReloKind(), cg()),
                   __FILE__, __LINE__, getNode());
                break;
+            case TR_MethodPointer:
+               if (getNode() && getNode()->getInlinedSiteIndex() == -1)
```

Comment:
Thanks for the suggestion. Modified accordingly and fixed other similar conditions in the code.